{
    "url": "https://api.github.com/repos/spack/spack/issues/18876",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/18876/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/18876/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/18876/events",
    "html_url": "https://github.com/spack/spack/issues/18876",
    "id": 706701565,
    "node_id": "MDU6SXNzdWU3MDY3MDE1NjU=",
    "number": 18876,
    "title": "Installation issue: py-psycopg2 fails to build with python@3.8",
    "user": {
        "login": "Sinan81",
        "id": 2924561,
        "node_id": "MDQ6VXNlcjI5MjQ1NjE=",
        "avatar_url": "https://avatars.githubusercontent.com/u/2924561?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/Sinan81",
        "html_url": "https://github.com/Sinan81",
        "followers_url": "https://api.github.com/users/Sinan81/followers",
        "following_url": "https://api.github.com/users/Sinan81/following{/other_user}",
        "gists_url": "https://api.github.com/users/Sinan81/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/Sinan81/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/Sinan81/subscriptions",
        "organizations_url": "https://api.github.com/users/Sinan81/orgs",
        "repos_url": "https://api.github.com/users/Sinan81/repos",
        "events_url": "https://api.github.com/users/Sinan81/events{/privacy}",
        "received_events_url": "https://api.github.com/users/Sinan81/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 446616547,
            "node_id": "MDU6TGFiZWw0NDY2MTY1NDc=",
            "url": "https://api.github.com/repos/spack/spack/labels/build-error",
            "name": "build-error",
            "color": "ff3300",
            "default": false,
            "description": null
        }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 1,
    "created_at": "2020-09-22T21:21:18Z",
    "updated_at": "2020-09-24T19:41:02Z",
    "closed_at": "2020-09-24T19:41:02Z",
    "author_association": "CONTRIBUTOR",
    "active_lock_reason": null,
    "body": "\r\n### Steps to reproduce the issue\r\nIt seems this package fails to build with latest python, 3.8.5. It builds just fine with `python@3.7.8`\r\n\r\n```console\r\n$ spack install py-psycopg2\r\n...\r\n==> Installing py-psycopg2\r\n==> No binary for py-psycopg2 found: installing from source\r\n==> Using cached archive: /mnt/local/spack/var/spack/cache/_source-cache/archive/ec/eccf962d41ca46e6326b97c8fe0a6687b58dfc1a5f6540ed071ff1474cea749e.tar.gz\r\n==> py-psycopg2: Executing phase: 'build'\r\n==> Error: ProcessError: Command exited with status 1:\r\n    '/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/python-3.8.5-vaawtghjfbpb5nws2xfrrrm3252khyuq/bin/python3.8' '-s' 'setup.py' '--no-user-cfg' '\r\nbuild'\r\n\r\n2 errors found in build log:\r\n     54    running build_ext\r\n     55    building 'psycopg2._psycopg' extension\r\n     56    creating build/temp.linux-x86_64-3.8\r\n     57    creating build/temp.linux-x86_64-3.8/psycopg\r\n     58    /mnt/local/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -DPSYCOPG_DEFAULT_PYDATETIME=1 \r\n           -DPSYCOPG_VERSION=2.7.5 (dt dec pq3 ext lo64) -DPG_VERSION_NUM=110002 -DHAVE_LO64=1 -I/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/p\r\n           ython-3.8.5-vaawtghjfbpb5nws2xfrrrm3252khyuq/include/python3.8 -I. -I/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/postgresql-11.2-6s\r\n           4527k55r2n52g5deyugy437ozlilfb/include -I/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/postgresql-11.2-6s4527k55r2n52g5deyugy437ozlil\r\n           fb/include/server -c psycopg/psycopgmodule.c -o build/temp.linux-x86_64-3.8/psycopg/psycopgmodule.o -Wdeclaration-after-statement\r\n     59    psycopg/psycopgmodule.c: In function 'psyco_is_main_interp':\r\n  >> 60    psycopg/psycopgmodule.c:689:18: error: dereferencing pointer to incomplete type 'PyInterpreterState' {aka 'struct _is'}\r\n     61         while (interp->next)\r\n     62                      ^~\r\n     63    \r\n     64    It appears you are missing some prerequisite to build the package from source.\r\n     65    \r\n     66    You may install a binary package by installing 'psycopg2-binary' from PyPI.\r\n     67    If you want to install psycopg2 from source, please install the packages\r\n     68    required for the build and try again.\r\n     69    \r\n     70    For further information please check the 'doc/src/install.rst' file (also at\r\n     71    <http://initd.org/psycopg/docs/install.html>).\r\n     72    \r\n  >> 73    error: command '/mnt/local/spack/lib/spack/env/gcc/gcc' failed with exit status 1\r\n\r\nSee build log for details:\r\n  /cache/sbulut/spack-stage/spack-stage-py-psycopg2-2.7.5-cyfelqawl25prflqej3ntw5jfwapu3tf/spack-build-out.txt\r\n```\r\n\r\n### Information on your system\r\n\r\n```\r\n$ spack debug report\r\n* **Spack:** 0.15.4-1050-b272e00\r\n* **Python:** 3.6.8\r\n* **Platform:** linux-centos7-skylake\r\n```\r\n\r\n### Additional information\r\n\r\n[spack-build-out.txt](https://github.com/spack/spack/files/5264050/spack-build-out.txt)\r\n\r\n\r\n### General information\r\n\r\n- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform\r\n- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers\r\n- [x] I have uploaded the build log and environment files\r\n- [x] I have searched the issues of this repo and believe this is not a duplicate\r\n",
    "performed_via_github_app": null
}